modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -149,43 +192,69 @@ export default class GeoJsonLayer extends CompositeLayer {
     }
   }
 
-  /* eslint-disable complexity */
+  /* eslint-disable complexity, max-statements */
```

Comment:
This function is way too big. My suggestion is that we define the prop mapping in a constant, something like:

```js
const POINT_ICON_PROPS = {
  iconAtlas: 'iconAtlas',
  iconMapping: 'iconMapping',
  getPointAngle: 'getAngle',
  getFillColor: 'getColor',
  getIcon: 'getIcon'
};
```

Then we can use this in multiple places:

```js
function getDefaultProps(mapping, LayerClass) {
  const result = {};
  for (const key in mapping) {
    target[key] = LayerClass.defaultProps[mapping[key]];
  }
  return result;
}

const defaultProps = {
  ...getDefaultProps(POINT_ICON_PROPS, IconLayer),
  ...getDefaultProps(POINT_TEXT_PROPS, TextLayer),
  ...getDefaultProps(POINT_CIRCLE_PROPS, ScatterplotLayer),
  //...
};
```

```js
function getSubLayerProps(layer, mapping) {
  const props = {};
  for (const key in mapping) {
    let value = layer.props[key];
    if (key.startsWith('get')) {
      value = layer.getSubLayerAccessor(value);
    }
    result[mapping[key]] = value;
  }
  return result;
}
```